tests/cnf/core/network/metallb/tests/bgp-remote-as-dynamic.go

```diff
@@ -0,0 +1,224 @@
+					metallbenv.DefineIterationParams(
+						ipv4metalLbIPList, ipv6metalLbIPList, ipv4NodeAddrList, ipv6NodeAddrList, netparam.IPV4Family)
+				Expect(err).ToNot(HaveOccurred(), "Fail to set iteration parameters")
+			})
+
+			AfterEach(func() {
+				By("Clean metallb operator and test namespaces")
+				resetOperatorAndTestNS()
+			})
+
+			It("Verify the establishment of an eBGP adjacency using neighbor peer remote-as external",
+				reportxml.ID("76821"), func() {
+					By("Step up test cases with Frr Node AS 64500 and external Frr AS 64501")
+					frrk8sPods, frrPod := setupBGPRemoteASTestCase(addressPool, hubIPv4ExternalAddresses, externalAdvertisedIPv4Routes,
+						externalAdvertisedIPv6Routes, dynamicASeBGP, tsparams.RemoteBGPASN)
+
+					By("Checking that BGP session is established and up")
+					verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+					By("Validating external FRR AS number received on the FRR nodes")
+					Eventually(func() int {
+						remoteAS, err := frr.FetchBGPRemoteAS(frrk8sPods, ipv4metalLbIPList[0])
+						Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP remote AS")
+
+						return remoteAS
+					}, 60*time.Second, 5*time.Second).Should(Equal(tsparams.RemoteBGPASN),
+						fmt.Sprintf("The remoteASN doesnt not match the expected AS: %d", tsparams.RemoteBGPASN))
+				})
+
+			It("Verify the establishment of an iBGP adjacency using neighbor peer remote-as internal",
+				reportxml.ID("76822"), func() {
+					By("Step up test cases with Frr Node AS 64500 and external Frr AS 64500")
+					frrk8sPods, frrPod := setupBGPRemoteASTestCase(addressPool, hubIPv4ExternalAddresses, externalAdvertisedIPv4Routes,
+						externalAdvertisedIPv6Routes, dynamicASiBGP, tsparams.LocalBGPASN)
+
+					By("Checking that BGP session is established and up")
+					verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+					By("Validating external FRR AS number received on the FRR nodes")
+					Eventually(func() int {
+						remoteAS, err := frr.FetchBGPRemoteAS(frrk8sPods, ipv4metalLbIPList[0])
+						Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP remote AS")
+
+						return remoteAS
+					}, 60*time.Second, 5*time.Second).Should(Equal(tsparams.LocalBGPASN),
+						fmt.Sprintf("The remoteASN doesnt not match the expected AS: %d", tsparams.LocalBGPASN))
+				})
+
+			It("Verify the failure to establish a iBGP adjacency with a misconfigured external FRR pod",
+				reportxml.ID("76825"), func() {
+					By("Step up test cases with Frr Node AS 64500 and misconfigured iBGP external Frr AS 64501")
+					frrk8sPods, frrPod := setupBGPRemoteASTestCase(addressPool, hubIPv4ExternalAddresses, externalAdvertisedIPv4Routes,
+						externalAdvertisedIPv6Routes, dynamicASiBGP, tsparams.RemoteBGPASN)
+
+					By("Checking that BGP session is down")
+					verifyMetalLbBGPSessionsAreDownOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
```

Comment:
maybe we should have a consistently here, because being down at the start of the test does not mean it would stay down

Reply:
I think its fine to leave as Eventual. The BGP peer is recreated in each test.  It waits 3 minutes to see if the peer will be established.

Reply:
OK after our call I changed it to consistently. It is checking for established and will fail after 60 seconds.